File: client/src/main/java/com/jraska/github/client/users/UserDetail.java
package com.jraska.github.client.users;

import android.support.annotation.NonNull;

import java.util.Collections;
import java.util.List;

public class UserDetail {
  @NonNull public final UserStats _basicStats;
  @NonNull public final List<Repo> _popularRepos;

  public UserDetail(@NonNull UserStats basicStats, @NonNull List<Repo> popularRepos) {
    _basicStats = basicStats;
    _popularRepos = Collections.unmodifiableList(popularRepos);
  }
}

File: feature-users/src/main/java/com/jraska/github/client/users/model/UserDetail.kt
package com.jraska.github.client.users.model

import java.util.Collections

internal class UserDetail(
  val user: User,
  val basicStats: UserStats?,
  popularRepos: List<RepoHeader>,
  contributedRepos: List<RepoHeader>
) {

  val popularRepos: List<RepoHeader> = Collections.unmodifiableList(popularRepos)
  val contributedRepos: List<RepoHeader> = Collections.unmodifiableList(contributedRepos)
}
